### PR TITLE
Bundle partitions

### LINKED ISSUE #27: bundle splits with dataset
This just means that there is a method to load the partitions from the devkit. This could be done with text files in the repo itself, or putting them in the AWS bucket.

### PR DESCRIPTION
**Pull request by una-dinosauria**
Bundles partition into the dataloader. The dataloader may now receive an additional argument `partitions`, and provides a boolean array that indicates which sensor measurements (currently, from the front camera), should be skipped. If `partitions` is `None`, then no frames are skipped at all.

Closes #27

### CODE REVIEW

File: pit30m/data/log_reader.py [hunk 1]
```diff
@@ -75,17 +75,53 @@ def __init__(
...
         self._map = Map() if map is None else map
         # TODO(julieta) Semantic version this
         self._index_version = index_version
+        self.partitions = set() if partitions is None else partitions
 
     def __repr__(self) -> str:
         return f"Pit30M Log Reader: {self._log_root_uri}"
 
+    @cached_property
+    def partitions_index(self) -> np.ndarray:
```
Comment (AndreiBarsan) [inline pit30m/data/log_reader.py:93]: Should this be called `partition_mask`?

File: pit30m/data/log_reader.py [hunk 2]
```diff
@@ -75,17 +75,53 @@ def __init__(
...
+        """
+        if not self.partitions:
+            n_sensor_measurements = len(self.get_cam_geo_index(CamName.MIDDLE_FRONT_WIDE))
+            return np.full(n_sensor_measurements, True)
+
+        partition_indices = self.partition_assigments
+        combined_indices = np.logical_and.reduce(partition_indices)
+        return combined_indices
+
+    @property
```
Comment (AndreiBarsan) [inline pit30m/data/log_reader.py:106]: cache?

File: pit30m/data/log_reader.py [hunk 3]
```diff
@@ -75,17 +75,53 @@ def __init__(
...
+    def partition_assigments(self):
+        """Fetches partition indices from S3 and converts them to boolean arrays according to the reader partition values"""
+        partitions_basepath = "s3://pit30m/partitions/"
```
Comment (AndreiBarsan) [inline pit30m/data/log_reader.py:109]: module level constant?

File: pit30m/data/partitions.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
+from __future__ import annotations
+
+from abc import abstractmethod
+from enum import Enum
+
+import numpy as np
+
+
+class PartitionEnum(Enum):
```
Comment (AndreiBarsan) [inline pit30m/data/partitions.py:9]: Nit: Unless you have a strong preference, it's fine to not name this `...Enum`

**Review by AndreiBarsan — APPROVED** [review]
Nice! Thank you! LGTM with minor comments

File: pit30m/data/log_reader.py (hunk 1 above)
Comment (una-dinosauria) [reply pit30m/data/log_reader.py]: Addressed in [0b0e9ab](https://github.com/pit30m/pit30m/pull/35/commits/0b0e9ab1589ad35c32c384b72f58b1b9504fd763)

File: pit30m/data/log_reader.py (hunk 3 above)
Comment (una-dinosauria) [reply pit30m/data/log_reader.py]: addressed in [3821752](https://github.com/pit30m/pit30m/pull/35/commits/38217529b1078c9edc2cfd8620df40288db5a63e)

File: pit30m/data/partitions.py (hunk 4 above)
Comment (una-dinosauria) [reply pit30m/data/partitions.py]: Good point! Addressed in [e92f984](https://github.com/pit30m/pit30m/pull/35/commits/e92f9842707ef920ff901e48725ef0249c958608)

File: pit30m/data/log_reader.py (hunk 2 above)
Comment (una-dinosauria) [reply pit30m/data/log_reader.py:111]: Yep! Added in [5b75a33](https://github.com/pit30m/pit30m/pull/35/commits/5b75a337ea74282ce5896d3a659e916b2c39b22a)

**Comment by AndreiBarsan** [comment]
Thank you for addressing the comments!!

**Pull request merged**